### ISSUE TITLE
Orchard azure web app stop working with sites.config file was changed by another program.

### ISSUE THREAD

**Issue by RichardSunMS** [OrchardCMS/Orchard#7742]
Hi, my web app stop working after running well couple of days with below exception in the log file.

And it will start to work after I saved the web.config file even without changing anything in it.

Then stop working again with another random days running well.

Any suggestions?

Thanks.

2017-06-15 17:00:19,120 [8] Orchard.Environment.DefaultOrchardHost - (null) - A tenant could not be started: Default
(null)
System.Configuration.ConfigurationErrorsException: The configuration file has been changed by another program. (D:\home\site\wwwroot\Config\Sites.config)
   at System.Configuration.BaseConfigurationRecord.EvaluateOne(String[] keys, SectionInput input, Boolean isTrusted, FactoryRecord factoryRecord, SectionRecord sectionRecord, Object parentResult)
   at System.Configuration.BaseConfigurationRecord.Evaluate(FactoryRecord factoryRecord, SectionRecord sectionRecord, Object parentResult, Boolean getLkg, Boolean getRuntimeObject, Object& result, Object& resultRuntimeObject)
   at System.Configuration.BaseConfigurationRecord.GetSectionRecursive(String configKey, Boolean getLkg, Boolean checkPermission, Boolean getRuntimeObject, Boolean requestIsHere, Object& result, Object& resultRuntimeObject)
   at Autofac.Configuration.SectionHandler.Deserialize(String configurationFile, String configurationSection)
   at Orchard.Environment.ShellBuilders.ShellContainerFactory.<>c__DisplayClass3_0.<CreateContainer>b__2(ContainerBuilder builder)
   at Autofac.Core.Lifetime.LifetimeScope.CreateScopeRestrictedRegistry(Object tag, Action`1 configurationAction)
   at Autofac.Core.Lifetime.LifetimeScope.BeginLifetimeScope(Object tag, Action`1 configurationAction)
   at Orchard.Environment.ShellBuilders.ShellContextFactory.CreateShellContext(ShellSettings settings)
   at Orchard.Environment.DefaultOrchardHost.<CreateAndActivateShells>b__31_1(ShellSettings settings)

**Comment by douwinga** [comment]
@RichardSunMS ever figure anything out about this issue?

We saw this same issue on one of our Orchard v.1.10.1.0 instances hosted on Azure Web Apps. It first happened on 06/06/17, but then happened again 06/28/17 and 06/29/17. This is an instance that has been running without issue since 03/26/17.

Stopping and starting the app pool does not fix the issue, however changing a setting in the Web.config file to get the app to just restart works (for how long, I am not sure).

We have a few other Orchard instances running in Azure Web Apps without any issues.

**Comment by v1r7u** [comment]
Hi, guys!
Same issue for our app in Azure: first happened 06/06/2017, then again 22th, 28th, 30th.
Restart won't help, fake change in Web.config bring the app up.

**Comment by agb-nementis** [comment]
Hello,

I have same issue after enabling SSL module on azure site version 1.10.2. Ranaming web.config put application back online. Also happened few days ago without any changes on the app.

**Comment by agb-nementis** [comment]
Just an update with a possible new hint. Not sure it is really related but my site has run without issues for 10 days then yesterday I re enabled application insights. After few hours the site had again the issue described before. Now I've removed Application Insights . Will update the issue tomorrow with site status.

**Comment by duginov** [comment]
Same happened with my website running on Orchard in Azure without problems since 2014. Touching web.config helped. Does anybody know how to detect the issue if/when it happens again? Azure dashboard reports "Good health" while I receive 404 - Server Error in '/' Application

**Comment by yechongbo** [comment]
facing this issue as well. Does this one https://stackoverflow.com/questions/44734788/azure-web-app-stop-working-with-sites-config-file-was-changed-by-another-program really helps?

**Comment by agb-nementis** [comment]
Done some tests: as suggested by yechongbo, adding WEBSITE_DYNAMIC_CACHE application setting with value 0 allowed me to enable/disable several modules without any issues (an activity that few days ago always caused the above error).
As soon as I removed the application setting, when I tried to enable a module, I got back the application error. So this seems a fix to an azure related issue. Hope it helps...
